src/helpers/domains.ts

```diff
@@ -0,0 +1,26 @@
+import { ethers } from 'ethers';
+
+const getSubnodeHash = (parentHash: string, labelHash: string): string => {
+	const calculatedHash = ethers.utils.keccak256(
+		ethers.utils.defaultAbiCoder.encode(
+			['bytes32', 'bytes32'],
+			[ethers.utils.arrayify(parentHash), ethers.utils.arrayify(labelHash)],
+		),
+	);
+
+	return calculatedHash;
+};
+
+export const getDomainId = (name: string): string => {
+	let hashReturn = ethers.constants.HashZero;
+
+	if (name === '' || undefined || null) {
+		return hashReturn;
+	}
+
+	const domains = name.split('.');
+	for (let i = 0; i < domains.length; i++) {
+		hashReturn = getSubnodeHash(hashReturn, ethers.utils.id(domains[i]));
+	}
+	return hashReturn;
```

Comment:
I know this was copied over from zNS-dapp`, but I'm just realising how weird the implementation is. Specifically the `if (name === '' || undefined || null)`, as this is just equivalent to `if (name === '')`.

```
if(!name.length) {
    return hashReturn;
}
```

Obviously not your fault there, but if you could refactor it that would be handy!

Reply:
this is totally removed, it is outdated method!!!!!